azext_iot/operations/digitaltwin.py

```diff
@@ -0,0 +1,227 @@
+    else:
+        target_interfaces = interface_list
+
+    for entity in target_interfaces:
+        interface_result = {'name': entity['name'], 'urn_id': entity['urn_id'], 'commands': {}}
+        interface_commands = []
+        found_commands = _pnp_interface_elements(cmd, entity['urn_id'], INTERFACE_COMMAND,
+                                                 repo_endpoint, repo_id, repo_login)
+        for command in found_commands:
+            command.pop('@type', None)
+            interface_commands.append(command)
+        interface_result['commands'] = interface_commands
+        result.append(interface_result)
+    return {'interfaces': result}
+
+
+def iot_digitaltwin_properties_list(cmd, device_id, interface=None,
+                                    hub_name=None, resource_group_name=None, login=None):
+    result = []
+    device_interfaces = _iot_digitaltwin_interface_list(cmd, device_id, hub_name, resource_group_name, login)
+    interface_list = _get_device_default_interface_dict(device_interfaces)
+    interface_count = len([k for k in interface_list if k['name'] == interface])
+
+    if interface and interface_count == 0:
+        raise CLIError('Target interface is not implemented by the device!')
+
+    for entity in interface_list:
+        properties = []
+        interface_data = device_interfaces['interfaces'].get(entity['name'])
+        if interface_data:
+            properties = interface_data.get('properties')
+        result.append({'name': entity['name'], 'urn_id': entity['urn_id'], 'properties': properties})
+
+    if interface:
+        return {'interfaces': [item for item in result if item['name'] == interface]}
+    return {'interfaces': result}
+
+
+# pylint: disable=too-many-locals
+def iot_digitaltwin_invoke_command(cmd, interface, device_id, command_name, schema=False, command_payload=None,
+                                   timeout=10, repo_endpoint=PNP_ENDPOINT, repo_id=None, hub_name=None,
+                                   resource_group_name=None, login=None, repo_login=None):
+    device_interfaces = _iot_digitaltwin_interface_list(cmd, device_id, hub_name, resource_group_name, login)
+    interface_list = _get_device_default_interface_dict(device_interfaces)
+
+    target_interface = [item for item in interface_list if item['name'] == interface]
+    interface_count = len(target_interface)
+    if interface_count == 0:
+        raise CLIError('Target interface is not implemented by the device!')
+
+    found_commands = _pnp_interface_elements(cmd, target_interface[0]['urn_id'], INTERFACE_COMMAND,
+                                             repo_endpoint, repo_id, repo_login)
+
+    if command_name not in [c.get('name') for c in found_commands if c.get('name') == command_name]:
+        raise CLIError('Command \'{}\' is not defined in the target interface.'.format(command_name))
+
+    target_command = [c for c in found_commands if c.get('name') == command_name]
+    if schema:
```

Comment:
Do we still need to keep this --schema parameter? What is this trying to do?

Reply:
The purpose of --schema (at least in private preview) was a shortcut to see the definition of a particular command or property.

Reply:
it will result the schema of command

Reply:
I am not sure if we should keep this to be honest.
For public preview, there are so many sources of model definition. I am assuming what we want to do here is, if user don't specify repo, get schema from public repo, and if so, get it from private repo.
But there are other sources as well, of which "get model definition from device itself" is what we we are going to enable on the UI. I am not sure if what works before still serves it well today, so I was trying to eliminating this complexity from CLI, or we will need to support a more sophiscated configuration of sources like our UI does. 
What do you say @iluican

Reply:
And in my personal opinion, we should simply remove it, since we are not supporting this --schema parameter for property, why only commands then.

Reply:
Implemented as per the mail conversation